Mathlib/GroupTheory/Perm/Cycle/Basic.lean

```diff
@@ -1978,3 +2080,213 @@ theorem _root_.Finset.sum_mul_sum_eq_sum_perm (hσ : σ.IsCycleOn s) (f g : ι 
 #align finset.sum_mul_sum_eq_sum_perm Finset.sum_mul_sum_eq_sum_perm
 
 end Finset
+
+theorem subtypePerm_apply_pow_of_mem {g : Perm α} {s : Finset α}
+    (hs : ∀ x : α, x ∈ s ↔ g x ∈ s) {n : ℕ} {x : α} (hx : x ∈ s) :
+    ((g.subtypePerm hs ^ n) (⟨x, hx⟩ : s) : α) = (g ^ n) x := by
+  revert x
+  induction n with
```

Comment:
We usually use `induction n generalizing x` instead of `revert x; induction n <;> intro x`

Reply:
That's indeed simpler; I'll adjust swiftly. Thanks!